pkg/autoscaler/autoscaler.go

```diff
@@ -254,6 +254,20 @@ var _ = Describe("Autoscaler should", framework.LabelAutoscaler, Serial, func()
 
 			framework.WaitForMachineSet(ctx, client, machineSet.GetName())
 
+			Eventually(func() (map[string]string, error) {
+				//checking for the keys of the old scalefromzero annotations before creating a machineautoscaler
+				ms, err := framework.GetMachineSet(context.TODO(), client, machineSet.GetName())
+				if err != nil {
+					return nil, err
+				}
+
+				return ms.Annotations, nil
+			}, framework.WaitMedium, pollingInterval).Should(SatisfyAll(
+				HaveKey(annotationsutil.CpuKeyDeprecated),
+				HaveKey(annotationsutil.MemoryKeyDeprecated),
+				HaveKey(annotationsutil.GpuCountKeyDeprecated),
```

Comment:
Should there not be 4 annotations? GPUType is missing?

Reply:
Hey, these are checking just for the old ones, and GpuType didn't exist at this time :) it was introduced in the newer upstream annotations